### PR TITLE
[TwigBridge] suggest Translation Component when TranslationExtension is used

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 4.3
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #31754
| License       | MIT
| Doc PR        | -

### CODE REVIEW

File: src/Symfony/Bridge/Twig/Extension/TranslationExtension.php [hunk 1]
```diff
@@ -49,11 +49,19 @@ public function __construct($translator = null, NodeVisitorInterface $translatio
...
     /**
-     * @deprecated since Symfony 4.2
+     * @return TranslatorInterface|null
      */
     public function getTranslator()
     {
-        @trigger_error(sprintf('The "%s()" method is deprecated since Symfony 4.2.', __METHOD__), E_USER_DEPRECATED);
+        if (null === $this->translator) {
+            if (!interface_exists(TranslatorInterface::class)) {
+                throw new \LogicException(sprintf('You cannot use the "%s" if the Translation Contracts are not available. Try running "composer require symfony/translation-contracts".', __CLASS__));
```
Comment (fabpot) [inline src/Symfony/Bridge/Twig/Extension/TranslationExtension.php]: What about recommending `symfony/translator` instead? If there are some usage of the Translation extension, it means that people want to use translation. So, they need the concrete implementation, not just the contracts.

**Review by fabpot — CHANGES_REQUESTED** [review]

File: src/Symfony/Bridge/Twig/Extension/TranslationExtension.php (hunk 1 above)
Comment (nicolas-grekas) [reply src/Symfony/Bridge/Twig/Extension/TranslationExtension.php]: Why not. What's the use case for using the extension without providing it a translator?

Comment (fabpot) [reply src/Symfony/Bridge/Twig/Extension/TranslationExtension.php]: The only use case is for third-party bundles using translations, but that must be compatible with projects not using translations. But anyway, the main usage is when people wants to use translations, in which case, we need to recommend the implementation.

Comment (nicolas-grekas) [reply src/Symfony/Bridge/Twig/Extension/TranslationExtension.php]: ACK, updated

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Pull request merged**